contracts/uniswap_exchange.vy

```diff
@@ -63,32 +79,31 @@ def addLiquidity(min_liquidity: uint256, max_tokens: uint256, deadline: timestam
     total_liquidity: uint256 = self.totalSupply
     if total_liquidity > 0:
         assert min_liquidity > 0
-        eth_reserve: uint256(wei) = self.balance - msg.value
+        eth_reserve: uint256 = as_unitless_number(self.balance - msg.value)
         token_reserve: uint256 = self.token.balanceOf(self)
-        token_amount: uint256 = msg.value * token_reserve / eth_reserve + 1
-        liquidity_minted: uint256 = msg.value * total_liquidity / eth_reserve
+        # Platform profit is in ppm so we divide the liquidity minted by a 1000
+        platform_profit: uint256 = self.sqrt(1000000*eth_reserve*token_reserve/self.previous_invariant)
+        platform_liquidity_minted: uint256 = (total_liquidity * platform_profit * (10000 - self.platform_fee)/(10000+10000*platform_profit-platform_profit*(10000 - self.platform_fee)))/1000
+        token_amount: uint256 = as_unitless_number(msg.value) * token_reserve / eth_reserve + 1
+        liquidity_minted: uint256 = as_unitless_number(msg.value) * (total_liquidity + platform_liquidity_minted) / eth_reserve
```

Comment:
Liquidity the person is getting is Is dependent on the total liquidity (which is the previous total liquidity plus shares minted to the platform owner).